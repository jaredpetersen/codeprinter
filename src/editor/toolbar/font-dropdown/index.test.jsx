import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import FontDropdown from './index';
import { DropdownItem } from 'reactstrap';

describe('Editor Toolbar FontDropdown', () => {
  it('renders without crashing', () => {
    const fonts = ['serif', 'sans-serif'];
    shallow(<FontDropdown fonts={fonts} active={fonts[0]} onSelect={() => {}} />);
  });

  it('renders all of the fonts as dropdown options', () => {
    const fonts = ['serif', 'sans-serif'];
    const tree = renderer.create(<FontDropdown fonts={fonts} active={fonts[1]} onSelect={() => {}} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('calls the onSelect function on selecting a dropdown option', () => {
    const fonts = ['serif', 'sans-serif'];
    const onSelect = jest.fn();
    const fontDropdown = shallow(<FontDropdown fonts={fonts} active={fonts[0]} onSelect={onSelect} />);

    fontDropdown
      .find(DropdownItem)
      .last()
      .simulate('click');

    expect(onSelect).toHaveBeenCalledTimes(1);
  });
});
