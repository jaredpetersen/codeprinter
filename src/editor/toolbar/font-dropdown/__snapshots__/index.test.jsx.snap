// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor Toolbar FontDropdown renders all of the fonts as dropdown options 1`] = `
<div
  className="dropdown"
  onKeyDown={[Function]}
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-label={null}
    className="dropdown-toggle-select btn btn-outline-dropdown-toggle-select btn-block"
    onClick={[Function]}
    style={
      Object {
        "fontFamily": "\\"sans-serif\\", monospace",
        "textAlign": "left",
      }
    }
    type="button"
  >
    sans-serif
  </button>
  <div
    aria-hidden={true}
    className="dropdown-menu"
    role="menu"
    tabIndex="-1"
  >
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      style={
        Object {
          "fontFamily": "\\"serif\\", monospace",
        }
      }
      tabIndex="0"
      type="button"
    >
      serif
    </button>
    <button
      className="dropdown-item active"
      onClick={[Function]}
      role="menuitem"
      style={
        Object {
          "fontFamily": "\\"sans-serif\\", monospace",
        }
      }
      tabIndex="0"
      type="button"
    >
      sans-serif
    </button>
  </div>
</div>
`;
